source/material_model/prescribed_viscosity.cc

```diff
@@ -0,0 +1,197 @@
+          }
+          prm.leave_subsection ();
+
+          prm.enter_subsection ("Viscosity function");
+          {
+            Functions::ParsedFunction<dim>::declare_parameters (prm, 3);
+          }
+          prm.leave_subsection ();
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+
+    template <int dim>
+    void
+    PrescribedViscosity<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection ("Prescribed viscosity");
+        {
+          AssertThrow( prm.get("Base model") != "prescribed viscosity",
+                       ExcMessage("You may not use ``prescribed viscosity'' as the base model for "
+                                  "a prescribed viscosity model.") );
+
+          // create the base model and initialize its SimulatorAccess base
+          // class; it will get a chance to read its parameters below after we
+          // leave the current section
+          base_model = create_material_model<dim>(prm.get("Base model"));
+          if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(base_model.get()))
+            sim->initialize_simulator (this->get_simulator());
+
+          prm.enter_subsection("Indicator function");
+          {
+            try
+              {
+                prescribed_viscosity_indicator_function.parse_parameters (prm);
+              }
+            catch (...)
+              {
+                std::cerr << "ERROR: FunctionParser failed to parse\n"
+                          << "\t'Prescribed viscosity.Indicator function'\n"
+                          << "with expression\n"
+                          << "\t'" << prm.get("Function expression") << "'";
+                throw;
+              }
+          }
+          prm.leave_subsection();
+
+          prm.enter_subsection("Viscosity function");
+          {
+            try
+              {
+                prescribed_viscosity_function.parse_parameters (prm);
+              }
+            catch (...)
+              {
+                std::cerr << "ERROR: FunctionParser failed to parse\n"
+                          << "\t'Prescribed viscosity.Viscosity function'\n"
+                          << "with expression\n"
+                          << "\t'" << prm.get("Function expression") << "'";
+                throw;
+              }
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+
+      /* After parsing the parameters for depth dependent, it is essential to parse
+      parameters related to the base model. */
+      base_model->parse_parameters(prm);
+      this->model_dependence = base_model->get_model_dependence();
+    }
+
+    template <int dim>
+    bool
+    PrescribedViscosity<dim>::
+    is_compressible () const
+    {
+      return base_model->is_compressible();
+    }
```

Comment:
Please move this function above the parse_parameters and declare_parameters functions. This is not a hard rule, but I would like to keep our convention in as many files as possible. Makes it easier to read.